src/main/java/org/broadinstitute/dsm/db/dao/participant/data/ParticipantDataDao.java

```diff
@@ -0,0 +1,145 @@
+        }
+        return (int) simpleResult.resultValue;
+
+    }
+
+    @Override
+    public int delete(int id) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult execResult = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_DELETE_DDP_PARTICIPANT_DATA)) {
+                stmt.setInt(1, id);
+                execResult.resultValue = stmt.executeUpdate();
+            }
+            catch (SQLException ex) {
+                execResult.resultException = ex;
+            }
+            return execResult;
+        });
+        if (results.resultException != null) {
+            throw new RuntimeException("Error deleting participant data with "
+                    + id, results.resultException);
+        }
+        return (int) results.resultValue;
+    }
+
+    @Override
+    public Optional<ParticipantDataDto> get(long id) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult execResult = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_PARTICIPANT_DATA_BY_ID)) {
+                stmt.setLong(1, id);
+                try(ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                         execResult.resultValue = new ParticipantDataDto(
+                                        rs.getInt(1),
+                                        rs.getString(2),
+                                        rs.getInt(3),
+                                        rs.getString(4),
+                                        rs.getString(5),
+                                        rs.getLong(6),
+                                        rs.getString(7)
+                         );
+                    }
+                }
+            }
+            catch (SQLException ex) {
+                execResult.resultException = ex;
+            }
+            return execResult;
+        });
+        if (results.resultException != null) {
+            throw new RuntimeException("Error getting participant data with "
+                    + id, results.resultException);
+        }
+        return Optional.ofNullable((ParticipantDataDto) results.resultValue);
+    }
+
+    public List<ParticipantDataDto> getParticipantDataByParticipantId(String participantId) {
+        List<ParticipantDataDto> participantDataDtoList = new ArrayList<>();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult execResult = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_PARTICIPANT_DATA_BY_PARTICIPANT_ID)) {
+                stmt.setString(1, participantId);
+                try(ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        participantDataDtoList.add(
+                                new ParticipantDataDto(
+                                        rs.getInt(1),
```

Comment:
I am not sure I like not knowing what column we are adding to the object…

Reply:
good point, will change them.